# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/PersistentBufferProvider.cpp
# Commit: e3051b96067a
# Full Hash: e3051b96067a480b020bd8c984a54491665c765e
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2021-11-11 15:25:44
# Regressor Bug: 1739448
# File Overlap Count: 1
# Description:
#   Bug 1739448 - Implement a prototype WebGL-backed Canvas2D. r=gfx-reviewers,nical,jrmuizel
#   
#   This mainly provides DrawTargetWebgl, which implements the subset of the DrawTarget
#   API necessary for integration with CanvasRenderingContext2D. It translates them to
#   suitable commands for its internal ClientWebGLContext, which then manages remoting
# ==============================================================================

diff -r 66c30950518d -r e3051b96067a gfx/layers/PersistentBufferProvider.cpp
--- a/gfx/layers/PersistentBufferProvider.cpp	Thu Nov 11 07:16:57 2021 +0000
+++ b/gfx/layers/PersistentBufferProvider.cpp	Thu Nov 11 07:16:58 2021 +0000
@@ -96,6 +96,21 @@
   return provider.forget();
 }
 
+PersistentBufferProviderAccelerated::PersistentBufferProviderAccelerated(
+    DrawTarget* aDt)
+    : PersistentBufferProviderBasic(aDt) {
+  MOZ_COUNT_CTOR(PersistentBufferProviderAccelerated);
+}
+
+PersistentBufferProviderAccelerated::~PersistentBufferProviderAccelerated() {
+  MOZ_COUNT_DTOR(PersistentBufferProviderAccelerated);
+}
+
+ClientWebGLContext* PersistentBufferProviderAccelerated::AsWebgl() {
+  return (ClientWebGLContext*)mDrawTarget->GetNativeSurface(
+      NativeSurfaceType::WEBGL_CONTEXT);
+}
+
 // static
 already_AddRefed<PersistentBufferProviderShared>
 PersistentBufferProviderShared::Create(gfx::IntSize aSize,
@@ -184,15 +199,6 @@
   Destroy();
 }
 
-LayersBackend PersistentBufferProviderShared::GetType() {
-  if (mKnowsCompositor->GetCompositorBackendType() ==
-      LayersBackend::LAYERS_WR) {
-    return LayersBackend::LAYERS_WR;
-  } else {
-    return LayersBackend::LAYERS_CLIENT;
-  }
-}
-
 bool PersistentBufferProviderShared::SetKnowsCompositor(
     KnowsCompositor* aKnowsCompositor) {
   MOZ_ASSERT(aKnowsCompositor);